### PR TITLE
WIP: Added documentation as part of investigating Kafka-Journal

### PR DESCRIPTION
**Pull request by rtar**

### CODE REVIEW

File: eventual-cassandra/src/main/scala/com/evolutiongaming/kafka/journal/eventual/cassandra/JournalHead.scala [hunk 1]
```diff
@@ -6,7 +6,13 @@ import com.evolutiongaming.kafka.journal.{DeleteTo, PartitionOffset, SeqNr}
 import com.evolutiongaming.scassandra.syntax._
 import com.evolutiongaming.scassandra.{DecodeRow, EncodeRow}
 
-
+/** Represent metadata of the latest journal event saved to Cassandra.
```
Comment (t3hnar) [inline eventual-cassandra/src/main/scala/com/evolutiongaming/kafka/journal/eventual/cassandra/JournalHead.scala:9]: > Represent metadata of the latest journal

-> 

> Represent metadata of particular journal including info about last store event

wdyt?

File: eventual-cassandra/src/main/scala/com/evolutiongaming/kafka/journal/eventual/cassandra/JournalHead.scala [hunk 2]
```diff
@@ -6,7 +6,13 @@ import com.evolutiongaming.kafka.journal.{DeleteTo, PartitionOffset, SeqNr}
...
+  *
+  * @param partitionOffset the offset where the event is located in Kafka
+  * @param segmentSize the last segmentSize used to write a journal
```
Comment (t3hnar) [inline eventual-cassandra/src/main/scala/com/evolutiongaming/kafka/journal/eventual/cassandra/JournalHead.scala:12]: it's not last, it's constant for life of journal

File: eventual-cassandra/src/main/scala/com/evolutiongaming/kafka/journal/eventual/cassandra/JournalHead.scala [hunk 3]
```diff
@@ -6,7 +6,13 @@ import com.evolutiongaming.kafka.journal.{DeleteTo, PartitionOffset, SeqNr}
 import com.evolutiongaming.scassandra.syntax._
 import com.evolutiongaming.scassandra.{DecodeRow, EncodeRow}
 
-
+/** Represent metadata of the latest journal event saved to Cassandra.
+  *
+  * @param partitionOffset the offset where the event is located in Kafka
```
Comment (t3hnar) [inline eventual-cassandra/src/main/scala/com/evolutiongaming/kafka/journal/eventual/cassandra/JournalHead.scala:11]: it's not about event only, but any relevant record to this journal, and there are records like:

```
Append(events)
Delete
Purge
Mark
```

File: eventual-cassandra/src/main/scala/com/evolutiongaming/kafka/journal/eventual/cassandra/JournalHead.scala (hunk 2 above)
Comment (rtar) [reply eventual-cassandra/src/main/scala/com/evolutiongaming/kafka/journal/eventual/cassandra/JournalHead.scala:12]: What is expected to happen when we switch to larger `segmentSize`? Would it only be used for new journals?

**Review by t3hnar — APPROVED** [review]

**Review by t3hnar — APPROVED** [review]

**Pull request merged**